conda_smithy/templates/travis.yml.tmpl

```diff
@@ -1,76 +1,131 @@
 # This file was generated automatically from conda-smithy. To update this configuration,
 # update the conda-forge.yml and/or the recipe/meta.yaml.
 # -*- mode: yaml -*-
-
+{%- set platformset = [] %}
+{%- for _, platform, _ in configs %}
+  {%- set pfarchless = platform.split('-')[0] %}
+  {%- if pfarchless not in platformset %}
+    {%- do platformset.append(pfarchless) %}
+  {%- endif %}
+{%- endfor %}
 language: generic
 
-os: osx
-osx_image: {{ travis.get('osx_image', 'xcode9.4') }}
-
-
-{% block env -%}
-{% if configs[0] or travis.secure -%}
+{% if travis.secure -%}
 env:
-  {%- if configs[0] %}
-  matrix:
-    {%- for config, _, _ in configs %}
-    - CONFIG={{ config }}
-    {%- endfor %}
-  {%- endif %}
-{##}
-  {%- if travis.secure %}
   global:
     {%- for name, hashed_secure in travis.secure | dictsort %}
     # The {{ name }} secure variable. This is defined canonically in conda-forge.yml.
     - secure: "{{ hashed_secure }}"
     {%- endfor %}
-  {%- endif %}
 {%- endif %}
-{% endblock %}
+
+{% if configs[0] -%}
+matrix:
+  include:
+    {%- for config, platform, upload in configs | sort %}
+    - env: CONFIG={{ config }} UPLOAD_PACKAGES={{ upload }} platform={{ platform }}
+    {%- if platform.startswith("osx") %}
+      os: osx
+      osx_image: xcode9.4
+    {%- elif platform.startswith("linux-ppc64le") %}
+      os: linux-ppc64le
+    {%- endif %}
+
+    {%- endfor %}
+{%- endif %}
 
 before_install:
+{%- if 'linux' in platformset %}
+    - |
+      echo ""
+      echo "Maybe install python for ${platform}"
+      if [[ ${platform} =~ .*linux.* ]]; then
+        sudo apt-get update
+        sudo apt-get install python3 python3-pip
+        sudo pip3 install --upgrade pip
+        sudo pip3 install setuptools shyaml
+
+        which python3
+        # force having a python symlink
+        sudo ln -fs /usr/bin/python3 /bin/python
```

Comment:
See https://github.com/conda-forge/conda-smithy/issues/1020#issuecomment-464512877

Reply:
this was mostly to try and get a `python` around to run the fast finish script.  tried many iterations, nothing worked.